test-utils/contracts/alice/src/lib.rs

```diff
@@ -31,164 +26,159 @@ impl Alice {
-        }
+    #[no_mangle]
+    unsafe fn ping(arg_len: u32) -> u32 {
+        rusk_abi::wrap_query(arg_len, |()| STATE.ping())
+    }
 
-        pub fn withdraw_obfuscated(
-            &mut self,
-            message: Message,
-            message_address: StealthAddress,
-            change: Message,
-            change_address: StealthAddress,
-            output: Note,
-            proof: Vec<u8>,
-        ) {
-            let call = Call::withdraw_from_obfuscated(
+    #[no_mangle]
+    unsafe fn withdraw(arg_len: u32) -> u32 {
+        rusk_abi::wrap_transaction(arg_len, |(value, note, proof)| {
+            STATE.withdraw(value, note, proof)
+        })
+    }
+
+    #[no_mangle]
+    unsafe fn withdraw_obfuscated(arg_len: u32) -> u32 {
+        rusk_abi::wrap_transaction(
+            arg_len,
+            |(
                 message,
                 message_address,
                 change,
                 change_address,
                 output,
                 proof,
-            );
-            let call = Transaction::from_canon(&call);
-            let transfer = self.transfer;
-
-            dusk_abi::transact_raw(self, &transfer, &call, 0)
-                .expect("Failed to withdraw obfuscated!");
-        }
-
-        pub fn withdraw_to_contract(&mut self, to: ContractId, value: u64) {
-            let call = Call::withdraw_from_transparent_to_contract(to, value);
-            let call = Transaction::from_canon(&call);
-            let transfer = self.transfer;
-
-            dusk_abi::transact_raw(self, &transfer, &call, 0)
-                .expect("Failed to withdraw");
-        }
+            )| {
+                STATE.withdraw_obfuscated(
+                    message,
+                    message_address,
+                    change,
+                    change_address,
+                    output,
+                    proof,
+                )
+            },
+        )
     }
 
-    fn query(bytes: &mut [u8; PAGE_SIZE]) -> Result<(), CanonError> {
-        let mut source = Source::new(&bytes[..]);
-
-        let _contract = Alice::decode(&mut source)?;
-        let qid = u8::decode(&mut source)?;
-
-        match qid {
-            rusk_abi::PAYMENT_INFO => {
-                let ret = PaymentInfo::Any(None);
-
-                let r = {
-                    // return value
-                    let wrapped_return = ReturnValue::from_canon(&ret);
-
-                    let mut sink = Sink::new(&mut bytes[..]);
-
-                    wrapped_return.encode(&mut sink)
-                };
-
-                Ok(r)
-            }
-
-            _ => panic!(""),
-        }
+    #[no_mangle]
+    unsafe fn withdraw_to_contract(arg_len: u32) -> u32 {
+        rusk_abi::wrap_transaction(arg_len, |(to, value)| {
+            STATE.withdraw_to_contract(to, value)
+        })
     }
 
     #[no_mangle]
-    fn q(bytes: &mut [u8; PAGE_SIZE]) {
-        let _ = query(bytes);
+    unsafe fn payment_info(arg_len: u32) -> u32 {
```

Comment:
Payment info is not "supposed to" be introduced like this in the contract. I made a macro specifically for this reason. `rusk_abi::payment_info`.